### ISSUE TITLE
Crates using `license-file` get an empty license section in the dropdown

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/docs.rs#1497]
Example: https://docs.rs/serde-util/0.1.0/serde_util/
![image](https://user-images.githubusercontent.com/81079/133647371-e12ad716-1a91-419b-bc93-0ab75f63b5c2.png)

On crates.io this shows as "non-standard":
![image](https://user-images.githubusercontent.com/81079/133647463-01544483-31d5-4277-8d62-8faec04b415f.png)

**Comment by AndreLaranjeira** [comment]
From what I analysed, this is caused by how the information for the crate's license is initialized.

In the code for crates.io, whenever a new version for the crate is published, the [crate's metadata is parsed](https://github.com/rust-lang/crates.io/blob/3f67acf73e46aa6aa824719063a5e39e882635b0/src/controllers/krate/publish.rs#L61) and that information is used to [instantiate a NewVersion struct](https://github.com/rust-lang/crates.io/blob/3f67acf73e46aa6aa824719063a5e39e882635b0/src/controllers/krate/publish.rs#L158). When a NewVersion struct is created, the [call to the validate_license method](https://github.com/rust-lang/crates.io/blob/2e744af8573f90b9706fa72ccffe5a52f91e2fdc/src/models/version.rs#L144) checks the license file argument and sets the license field to "non-standard" if that field was previously empty and the license file is not empty.

In the code for docs.rs, when the metadata for the root package is generated by [this call to cargo](https://github.com/rust-lang/docs.rs/blob/17de2309726801404cdc8a6ef3143456eeeecf9c/src/utils/cargo_metadata.rs#L18), it appears the metadata for all the project's packages is simply [deserialized using serde_json](https://github.com/rust-lang/docs.rs/blob/17de2309726801404cdc8a6ef3143456eeeecf9c/src/utils/cargo_metadata.rs#L26) into a vector of Package structs, [which do not have a field for license file](https://github.com/rust-lang/docs.rs/blob/17de2309726801404cdc8a6ef3143456eeeecf9c/src/utils/cargo_metadata.rs#L47).

Maybe the aforementioned serde_json deserialization logic for Package could be altered to set the license field to "non-standard" instead of leaving it empty when the license_file field is not empty and that would solve the problem?

**Comment by Nemo157** [comment]
Yep, that sounds about what I would expect to be required. That will fix it for new builds, so would probably also pay to handle the license being `None` in [the template](https://github.com/rust-lang/docs.rs/blob/17de2309726801404cdc8a6ef3143456eeeecf9c/templates/rustdoc/topbar.html#L41) for existing builds by putting a default there (can this also just be `non-standard`, or is there another reason the license would be `None`?).

**Comment by AndreLaranjeira** [comment]
@Nemo157 You raise a good point. Since docs.rs only hosts the documentation for crates published on crates.io and [publishing a crate on crates.io requires a license or license-file](https://doc.rust-lang.org/cargo/reference/publishing.html#before-publishing-a-new-crate), then I _think_ the license field could just default to `non-standard` both on the template (for existing builds) and on the serde_json deserialization logic (for new builds) because any crate documented on docs.rs should have a license or license-file.

Having said that, I'm fairly new to the Rust community, so I'm not certain if all the past editions of Rust had a license requirement for publishing to crates.io.
